superset/views/core.py

```diff
@@ -1263,16 +1263,8 @@ def _set_dash_metadata(dashboard, data):
 
         if "timed_refresh_immune_slices" not in md:
             md["timed_refresh_immune_slices"] = []
-
         if "filter_scopes" in data:
-            md.pop("filter_immune_slices", None)
-            md.pop("filter_immune_slice_fields", None)
             md["filter_scopes"] = json.loads(data.get("filter_scopes", "{}"))
```

Comment:
```suggestion
            md["filter_scopes"] = json.loads(data["filter_scopes"])
```
not yours, but might as well clean this up

Reply:
there might be many dashboards holding NULL value in `data.get('filter_scopes')`.

Reply:
Since, this is inside the if statement `if "filter_scopes" in data:`, we don't need to use `.get` because we know the key is there. For the case you mentioned, I think this should be:
```python
md["filter_scopes"] = json.loads(data["filter_scopes"] or "{}")
```

For explanation, 
```python
data = {"filter_scopes": None}
print(data.get("filter_scopes", "{}")) # prints None
```

Reply:
you are right! added fix. thank you!